src/SSHLibrary/abstractclient.py

```diff
@@ -357,7 +359,7 @@ def _read_until(self, matcher, expected, timeout=None):
             if matcher(output):
                 return output
         raise SSHClientException("No match found for '%s' in %s\nOutput:\n%s."
-                                 % (expected, timeout, output))
+                                 % (self._decode(expected), timeout, output))
```

Comment:
`expected` should be passed as Unicode to avoid decoding it here. This is related to my previous comment.